includes/class-wc-coupon.php

```diff
@@ -415,7 +444,7 @@ public function get_discount_amount( $discounting_amount, $cart_item = null, $si
 	 * @throws WC_Data_Exception
 	 */
 	public function set_code( $code ) {
-		$this->data['code'] = apply_filters( 'woocommerce_coupon_code', $code );
+		$this->set_prop( 'code', apply_filters( 'woocommerce_coupon_code', $code ) );
```

Comment:
Might be good to introduce a formatting function for this filter so it's easier to call.